### ISSUE TITLE
Bad regexp in Chinese search

### ISSUE THREAD

**Issue by jwilk** [sphinx-doc/sphinx#2544]
Hi @enhao!

`sphinx/search/zh.py` contains this:

``` #!python
    latin1_letters = re.compile(r'\w+(?u)[\u0000-\u00ff]')
```

But the `\u` sequence is supported by the re module only since Python 3.3. In previous versions this regexp is equivalent to:

``` #!python
r'\w+(?u)[0-u]'
```

which is definitely not what you wanted.

But even in Python 3.3+, the regexp looks dubious to me. The variable name is `latin1_letters`, but the regexp matches a sequence of (not necessarily Latin 1) alphanumeric characters, followed by a Latin 1 character (not necessarily a letter).

**Comment by tk0miya** [comment]
We can merge the fix if someone confirms it works.

**Comment by jwilk** [comment]
BTW, this bug was found using [pydiatra](https://github.com/jwilk/pydiatra).

**Comment by AA-Turner** [comment]
Fixed by #5611
